partiql-logical-planner/src/lower.rs

```diff
@@ -543,118 +613,192 @@ impl<'ast> Visitor<'ast> for AstToLogical {
         self.siblings.pop();
 
         let mut benv = self.exit_benv();
-        assert_eq!(benv.len(), 1);
+        if self.eq_or_error(benv.len(), 1, "Expect benv.len() == 1") == Recurse::Stop {
+            return Recurse::Stop;
+        }
 
         let out = benv.pop().unwrap();
 
         let sink_id = self.plan.add_operator(BindingsOp::Sink);
         self.plan.add_flow(out, sink_id);
+        Recurse::Continue
     }
 
-    fn enter_query_set(&mut self, _query_set: &'ast QuerySet) {
+    fn enter_query_set(&mut self, _query_set: &'ast QuerySet) -> Recurse {
         self.enter_env();
 
         match _query_set {
-            QuerySet::SetOp(_) => todo!("QuerySet::SetOp"),
+            QuerySet::SetOp(_) => {
+                self.errors
```

Comment:
Might be worth a private helper method like your `true_or_error` method to encapsulate the creation of the `NotYetImplemented` error at these call sites.

Reply:
Yeah, there's some redundant code there. Probably will create a macro for the `NotYetImplemented` similar to the other errors.